webapi/src/main/scala/org/knora/webapi/routing/admin/ProjectsRouteADM.scala

```diff
@@ -173,18 +175,23 @@ class ProjectsRouteADM(routeData: KnoraRouteData)
     }
 
   /**
-   * returns a single project identified through iri
+   * Returns a single project identified through IRI or UUID
    */
-  private def getProjectByIri(): Route =
-    path(projectsBasePath / "iri" / Segment) { value =>
+  private def getProjectByIriOrUuid(routeResolver: String): Route =
```

Comment:
If you would like to implement it in this way, than you would need change a few things IMO:
1. create a proper type for the `routeResolver`.
2. Also, `routeResolver` is not the best name for it. Maybe something like IdSegment would be more descriptive. This new type would then be a sum type for UuidSegment and IriSegment.
3. To be consistent you would need to also add a `ShortcodeSegment` and a `ShortnameSegment`.

The point I want to emphasize is that the structure of the addition should be consistent with the rest of the implementation and not just a quick fix.

The other option would be to stick to the current design and simply add `getProjectByUuid`.

Reply:
Applied your suggestions in [f3a35b9](https://github.com/dasch-swiss/dsp-api/pull/2272/commits/f3a35b921b963eccc97171158ab646c29f652df5).